### PR TITLE
[ci] Clean up artifacts before/after jobs

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

We sometimes end up with stale wheel uploads from previous runs of a Buildkite agent. The result is that commit wheels are being overwritten from old build jobs - effectively breaking the wheel build logic.

Example:

This Agent: https://buildkite.com/organizations/ray-project/agents/4b955117-2f6c-4849-b703-3457daf69f89

- builds wheels (in post-wheels tests) for a35ebc945b
- and then runs both the Ray CPP worker and the Train + Tune tests in 6746e9f
- Usually these two tests shouldn't provide artifacts at all, but they do - these are the wheels from a35ebc945b though! Meaning these are uncleaned leftovers from the first build task.
- See here for proof of artifact upload: https://buildkite.com/ray-project/ray-builders-pr/builds/27622#d11bc514-ebd8-4e0c-a2ce-826b9bad27de

The solution is thus to always clean up the artifacts directory in the worker, i.e. `rm -rf /artifact-mount/*`

This PR adds two of such clean up instructions - once before commands are run and once after artifacts are uploaded. We can probably just do either, but it doesn't hurt to have both.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

**Comment by simon-mo** [comment]
Can we 
- shell check the scripts
- set -ex the scripts
- document the scripts

and to clarify this run in host machine right?

**Comment by krfricke** [comment]
Added docs and `set -xe` - the scripts are already included in shellcheck (tried it with `./scripts/format.sh`)

**Review by simon-mo — APPROVED** [review]

File: ci/travis/ci.sh [hunk 1]
```diff
@@ -415,8 +415,9 @@ build_wheels() {
         # Sync the directory to buildkite artifacts
         rm -rf /artifact-mount/.whl || true
         cp -r .whl /artifact-mount/.whl
+        chown -R 1000 /artifact-mount/.whl
```
Comment (krfricke) [inline ci/travis/ci.sh:418]:
```suggestion
        chmod -R 777 /artifact-mount/.whl
```

**Comment by krfricke** [comment]
Because of fs ownership issues (Docker creates root-owned files, but the command on the host container are executed as a regular using) e.g. the `/tmp/artifacts/.whl` directory is not deleted (but the contents are).
I'll merge this since the wheels themselves are deleted. The stale overwrite bug can happen until the current buildkite agents are cycled (if there are remaining root-owned wheels, they can't be deleted in the pre-command step).

**Pull request merged**